builder/qemu/builder.go

```diff
@@ -88,6 +88,8 @@ type Config struct {
 	DiskSize        uint       `mapstructure:"disk_size"`
 	DiskCache       string     `mapstructure:"disk_cache"`
 	DiskDiscard     string     `mapstructure:"disk_discard"`
+	DiskShrink      bool       `mapstructure:"disk_shrink"`
```

Comment:
Having both DiskShrink and DiskCompress seems a little strange to me since Compress would need both specified. What is your opinion of changing it to something like ```"DiskConvert = "shrink" or "compress"``` (that's just off the top of my head.

Also, looking at the qemu-img help it appears compress will only work with qcow format so there should be validity checks in the config checking. And based on this feedback could there be some tests added?